### ISSUE TITLE
Events are not properly cleaned up

### ISSUE THREAD

**Issue by firasdib** [dfcreative/pan-zoom#13]
You add events like this:
```
target.addEventListener('touchstart', initFn, hasPassive ? { passive: true } : false)
```

But remove them like this:
```
target.removeEventListener('touchstart', initFn)
```

When it should be:
```
target.removeEventListener('touchstart', initFn, hasPassive ? { passive: true } : false)
```

**Comment by firasdib** [comment]
I've fixed the memory leaks in Impetus here: https://github.com/chrisbateman/impetus/pull/41

But the author of that lib does not seem to be active anymore....

Can you use my fork instead? Or is there an alternative library?

**Comment by dy** [comment]
According to MDN https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener passive option does not matter on removal.

As for the impetus, if you have alternative suggestions let’s consider. Otherwise let’s wait for the merge.

**Comment by firasdib** [comment]
You're right, `{ passive: true }` does not need to be included. But if you specify capture, either by passing true/false as the third arg, or `{capture: true}`, you need to include it in the removal process.

> While addEventListener() will let you add the same listener more than once for the same type if the options are different, the only option removeEventListener() checks is the capture/useCapture flag.

**Comment by firasdib** [comment]
Btw, I just noticed the wheel library does not clean up after itself either.... This is also very bad.

**Comment by firasdib** [comment]
> Btw, I just noticed the wheel library does not clean up after itself either.... This is also very bad.
> 
> Edit:
> It turns out mouse-wheel is not cleaning up properly and its the one blocking all scrolling in our application.

**Comment by firasdib** [comment]
Excuse the mess.

My previous comment is not accurate, but something is weird. Something, somewhere, is not cleaning up properly.

**Comment by firasdib** [comment]
Bottom line: it was impetus and my fork fixes it. If the author does not merge it, should I publish it to npm and we use that instead?

**Comment by dy** [comment]
Let’s wait for a month. For now you can just use github as a dep. Should ping the author periodically

**Comment by firasdib** [comment]
@dy In that case I need to fork both your repo and his, and use that. Not ideal tbh.

**Comment by dy** [comment]
@firasdib why don't you just temporarily use your github fork as `"pan-zoom": "firasdib/pan-zoom"` dependency? If there is no response, I will include your fork as the main entry.

**Comment by firasdib** [comment]
@dy This is what I am doing now, but forks have a tendency to diverge... :-)
